pkg/activator/certificate/cache.go

```diff
@@ -40,23 +41,64 @@ import (
 type CertCache struct {
 	secretInformer v1.SecretInformer
 	logger         *zap.SugaredLogger
+	trustConfig    netcfg.Trust
 
-	certificate *tls.Certificate
-	TLSConf     tls.Config
+	certificate   *tls.Certificate
+	ClientTLSConf tls.Config
+	ServerTLSConf tls.Config
 
 	certificatesMux sync.RWMutex
+	caCerts         []*x509.Certificate
+}
+
+func (cr *CertCache) init() {
+	cr.ClientTLSConf.ServerName = certificates.LegacyFakeDnsName
```

Comment:
Use the new name?

Reply:
This is the right name to use here. 
This name appears in all client certificates and is currently used in all trust modes.

Reply:
I'm confused the feature track document says it should be verifying a `kn-user-*` certificate

Reply:
I was under the impression activator is always `kn-routing-0`.

Anyway, if I'm reading [this code](https://github.com/knative/networking/blob/14a2bd466ad64dc31c2941f79b1647eb9a2b5c63/pkg/certificates/reconciler/certificates.go#L107-L117) correctly, I think all the new certificates are getting reconciled to include the legacyfakednsname in addition to their new "kn-routing" or "kn-user" dns name.

Would that mean we can safely use the new names in this PR?

Reply:
This line is related to the activator's client info, right? That would be sent to the QueueProxy?

If that is the case, could we just skip the step of sending the legacyfakednsname and send the new name?

I think your [PR to queueproxy](https://github.com/knative/serving/pull/14063/files) will handle the case for the new `DataPlaneRoutingName("0")` anyway.

Reply:
Our new certificates include two sans:
- certificates.LegacyFakeDnsName
- kn-user-<namespace>
When the Activator is a client and approach queue, it needs to verify the server-side SAN
For all trust options, it verifies that certificates.LegacyFakeDnsName is included in the certificate. This is being done by the standard golang tls lib.

For trust = enabled, mutual, identity it also verifies that kn-user-<namespace> in the certificate. This is done by the verifySanConnection in transport.go.